src/node/blockstorage.cpp

```diff
@@ -785,19 +785,27 @@ bool ReadRawBlockFromDisk(std::vector<uint8_t>& block, const FlatFilePos& pos, c
     return true;
 }
 
-/** Store block on disk. If dbp is non-nullptr, the file is known to already reside on disk */
 FlatFilePos BlockManager::SaveBlockToDisk(const CBlock& block, int nHeight, CChain& active_chain, const CChainParams& chainparams, const FlatFilePos* dbp)
 {
     unsigned int nBlockSize = ::GetSerializeSize(block, CLIENT_VERSION);
     FlatFilePos blockPos;
-    if (dbp != nullptr) {
+    const auto position_known {dbp != nullptr}; // if dbp is not nullptr, then the file and block within it are known to already reside on disk
+    if (position_known) {
         blockPos = *dbp;
-    }
-    if (!FindBlockPos(blockPos, nBlockSize + 8, nHeight, active_chain, block.GetBlockTime(), dbp != nullptr)) {
+    } else {
+        // when known, blockPos.nPos points at the offset of the block data in the blk file. that already accounts for
+        // the serialization header present in the file (the 4 magic message start bytes + the 4 length bytes = 8 bytes).
+        // we add 8 only for new blocks since they will have the serialization header added when written to disk.
```

Comment:
Aid to reviewers: the 8 bytes mentioned are added on writing the block to disk in `blockstorage.cpp:WriteBlockToDisk()`.

Reply:
also, the position comes from `CChainState::LoadExternalBlockFile` when it's read in again. that position is past the serialization header already (it points directly at the block data) and thus why we don't need to add 8 with the old code `nBlockSize + 8`.